llmfoundry/common/fdiff.py

```diff
@@ -25,39 +25,38 @@ def __init__(self, diff_train_metrics=False, diff_eval_metrics=True):
 
     def batch_end(self, state: State, logger: Logger):
         if self.diff_train_metrics:
+            loss = state.loss.item()  # type: ignore
             if self.train_prev_loss:
-                logger.log_metrics({
-                    'loss/train/total_fdiff':
-                        state.loss.item() - self.train_prev_loss
-                })
-
-            self.train_prev_loss = state.loss.item()
+                logger.log_metrics(
+                    {'loss/train/total_fdiff': loss - self.train_prev_loss})
+            self.train_prev_loss = loss
 
             for k in self.train_prev_metric.keys():
                 logger.log_metrics({
                     f'metrics/train/{k}_fdiff':
-                        state.train_metric_values[k].item() -
+                        state.train_metric_values[k].item() -  # type: ignore
```

Comment:
The issue here is that `state.train_metric_values` is typed as a `Dict[str, float]` (https://github.com/mosaicml/composer/blob/dev/composer/core/state.py#L504), but this code assumes a tensor was put in. So is the `.item()` needed here? Is this masking an API misuse upstream?

Reply:
This callback isn't used super often
I think you just found a bug
:thankyou: